app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt

```diff
@@ -95,14 +107,37 @@ class ConversationListViewModel @Inject constructor(
         }
     }
 
-    private fun List<ConversationDetails>.toConversationsFoldersMap(): Map<ConversationFolder, List<GeneralConversation>> =
+    // TODO AR-1733
+    private fun prepareActivities(connections: List<ConnectionDetails>) =
+        connections.map {
+            NewActivity(eventType = EventType.ConnectRequest, it.conversation.toItem(
```

Comment:
same here not sure if those callbacks are needed : 
```

private fun prepareActivities(connections: List<ConnectionDetails>) =
        connections.map {
            NewActivity(eventType = EventType.ConnectRequest
                private = { privateConversation ->
                    PendingConnectionItem(
                        connectionInfo = ConnectionInfo(
                            it.connection.status.toString(), //TODO pass also user
                            it.connection.from
                        ),  it.conversation.toItem()
                    )
             
  
            ))
        }
```

would be sufficent I tihnk

Reply:
I think those are needed because I need to build different objects for `PendingConnectionItem` and `GeneralConversation`, thanks to that I don't need to duplicate so much logic